### ISSUE TITLE
Decide whether to implement custom Sampler for standard Training

### ISSUE THREAD

**Issue by MaxJoas** [jan-forest/autoencodix_package#81]
## Background.
Most of our PyTorch model architectures have BatchNorm Layers. So when by chance we have a sample_size and batch_size so that `len(data) % batch_size =1`. The variance in the BatchNorm calculation becomes 0, and we get a `DivisionByZeroError`. The standard solution would be to use drop_last in the PyTorch dataloader. However, we report our trainingdynamics (sample_ids, losses, latentspaces) on a per epoch basis (depending on the checkpoint interval). dropping samples would lead to inconsistent reporting and all sorts of problems when visualising our data. So we want to avoid this too.
This is an extreme edge case and depending on the assumption on user input, will occur with a probability of ~1%.

## Current solution
For our custom sampler for the XModalix this should not occur, and if it occurs we handle it by adding another random sample to this last batch). For the standard PyTorch DataLoader we check before creating the DataLoader if `len(data) % batch_size =1` in these cases we increase the batch_size by one and warn the user.

## Possible (better?) solution
We write a custom sampler here, too. I've already vibe-coded a sampler (in `src/autoencodix/data/_sampler`). The sampler will adjust the length of the last two batches if the last batch would have size 1 e.g if we have n_sample=513 and batch_size=128, instead of getting 128,128,128, 128 and 1, we would get 128,128,128, 64 and 65. To make this work with our custom DataSet classes, we would need to adjust the sampler for how our `__getitem__` method works and/or write a custom collate_fn.
In my opinion, the current fix is good enough, because the case is super rare and the custom sampler wouldn't solve the issue super differently than our quick fix.
